Notebooks for simulator sub-project

usmanwardag opened the pull request:
This pull-request will keep on adding notebooks throughout this project to discuss updates on various sub-modules.

dhuppenkothen commented:
Cool! But do we want notebooks in the library itself? Originally, we discussed having a separate repo for notebooks. Is that still on the table? At the very least, I think I'd suggest having a central folder at the top level of the stingray repo with all notebooks, rather than sprinkling them throughout the modules.

OrkoHunter commented:
I suspect there are going to be a lot of notebooks in the future. I'm in favor of pushing them over [here](https://github.com/StingraySoftware/notebooks).

matteobachetti commented:
Yes, I mostly agree with @dhuppenkothen and @OrkoHunter about using the other repository. An additional thing we might want to do is using binder.com to actually execute the notebooks online when possible.
Besides that, let's check the notebook content :)

dhuppenkothen commented:
I've heard good things about binder, though binder.com directs to a website that makes actual, physical binders. :) 
I think the one you mean is [this one](http://mybinder.org).

dhuppenkothen commented:
Ha! I'd forgotten we already had a repo for notebooks! Thanks, @OrkoHunter

matteobachetti commented:
Oops sorry for the wrong website name, I corrected my comment

OrkoHunter commented:
With the wrong website, I thought @matteobachetti is talking about printing and distributing notebooks, and that felt very weird :D

usmanwardag commented:
Great, I'd move the notebook there.

matteobachetti commented:
@usmanwardag My first and major comment is the follows. As it is now, when you look for the spectral bin in the CDF you just take the shortest distance from the CDF at a "tagged" frequency, corresponding to the center of a spectral bin. This assumes that bins are all equal. 
However, they are not. In general, they aren't equal in the spectrum, and - evidently - even less on the y-axis. 
So, instead of measuring distances from the bin center, it would be better, in my opinion, to consider if a given value falls into a range. In other words, when you compare the uniformly drawn value you should see if it falls between the CDF value corresponding to the start of a given bin and the CDF value at the end of that bin.
What do you think?

matteobachetti commented:
I guess that if you had a thousand bins this problem would be very limited. It is particularly evident when you have a small number of bins though, like in your notebook.

usmanwardag commented:
@matteobachetti thanks for the suggestion, that was indeed the problem. I have fixed that and moved to [here](https://github.com/StingraySoftware/notebooks/pull/2).

This PR can be closed now?

matteobachetti commented:
Yep. Thanks!
